examples/step-90/step-90.cc

```diff
@@ -0,0 +1,1271 @@
+                               dof_handler.n_dofs(),
+                               locally_relevant_dofs);
+    constraints.reinit(locally_owned_dofs, locally_relevant_dofs);
+
+    DoFTools::make_hanging_node_constraints(dof_handler, constraints);
+    constraints.close();
+    DoFTools::make_sparsity_pattern(dof_handler, dsp, constraints);
+
+    SparsityTools::distribute_sparsity_pattern(dsp,
+                                               locally_owned_dofs,
+                                               mpi_communicator,
+                                               locally_relevant_dofs);
+    global_matrix.reinit(locally_owned_dofs,
+                         locally_owned_dofs,
+                         dsp,
+                         mpi_communicator);
+
+    timer.stop();
+    pcout << "took (" << timer.wall_time() << "s)" << std::endl;
+  }
+
+  // @sect3{Assembly and surface accumulation}
+  // We use a MeshWorker to assemble the linear problem efficiently.
+  // This cell worker does not do anything for non-intersected cells.
+  template <int dim>
+  void LaplaceBeltramiSolver<dim>::assemble()
+  {
+    pcout << "Assembling ... " << std::flush;
+    TimerOutput::Scope t(computing_timer, "assembly");
+    Timer              timer;
+
+    const auto cell_worker =
+      [&](const typename DoFHandler<dim>::active_cell_iterator &cell,
+          ScratchData<dim>                                     &scratch_data,
+          CopyData<dim>                                        &copy_data) {
+        if (mesh_classifier.location_to_level_set(cell) ==
+              NonMatching::LocationToLevelSet::intersected &&
+            cell->is_locally_owned())
+          {
+            // Once we now that the cell is intersected, we construct the
+            // unfitted quadratures for the solutions FE space on the cell.
+            scratch_data.non_matching_fe_values.reinit(cell);
+            copy_data.reinit(cell);
+            copy_data.cell_matrix = 0;
+            copy_data.cell_rhs    = 0;
+            const std::optional<NonMatching::FEImmersedSurfaceValues<dim>>
+              &surface_fe_values =
+                scratch_data.non_matching_fe_values.get_surface_fe_values();
+            const std::vector<double> &JxW_surface =
+              surface_fe_values->get_JxW_values();
+
+            // The accumulation of the surface integrals, including the forcing,
+            // is performed here.
+            for (unsigned int q : surface_fe_values->quadrature_point_indices())
+              {
+                const Tensor<1, dim> &normal =
+                  surface_fe_values->normal_vector(q);
+
+                for (const unsigned int i : surface_fe_values->dof_indices())
+                  {
+                    copy_data.cell_rhs(i) +=
+                      right_hand_side.value(
+                        surface_fe_values->quadrature_point(q)) *
+                      surface_fe_values->shape_value(i, q) * JxW_surface[q];
```

Comment:
We like to multiply the equation from the left by the test function.
```suggestion
                      surface_fe_values->shape_value(i, q) *
                      right_hand_side.value(
                        surface_fe_values->quadrature_point(q)) *
                       JxW_surface[q];
```